File: python/860_柠檬水找零_easy.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        stack5, stack10 = [], []
        while bills:
            if bills[0]==5:
                stack5.append(bills[0])
                bills.pop(0)
            elif bills[0]==10:
                if stack5:
                    stack10.append(bills[0])
                    stack5.pop()
                    bills.pop(0)
                else:
                    return False
            else:
                if stack10 and stack5:
                    stack10.pop()
                    stack5.pop()
                    bills.pop(0)
                elif len(stack5)>2:
                    stack5 = stack5[:-3]
                    bills.pop(0)
                else:
                    return False
        return True

File: cpp/860_柠檬水找零_easy.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int five = 0, ten = 0;
        for(int bill : bills){
            if(bill == 5)
                five++;
            else if(bill == 10)
                if(five==0)
                    return false;
                else{
                    five--;
                    ten++;
                }
            else
                if(five>0 && ten>0){
                    five--;
                    ten--;
                }
                else if(ten == 0 && five > 2)
                    five -= 3;
                else 
                    return false;
        }
        return true;
    }
};